### ISSUE TITLE
Help with NBT related stuffs NEEDED

### ISSUE THREAD

**Issue by DefinitlyEvil** [DragonetMC/DragonProxy#43]

**Comment by jlirochon** [comment]
What is NBT ?

**Comment by MrPowerGamerBR** [comment]
NBT is a format Mojang uses for signs, entity data, tile entity data, etc.
Em 4 de jan de 2016 18:08, "Julien Lirochon" <notifications@github.com>
escreveu:

**Comment by jython234** [comment]
@DefinitlyEvil What NBT "stuffs"? Please be more specific.

**Comment by DefinitlyEvil** [comment]
@jython234 Tile entities can't spawn even after I added them in the chunk packet.

**Comment by jython234** [comment]
@DefinitlyEvil You sure you are using the right endianness?

**Comment by DefinitlyEvil** [comment]
@jython234 Yes, I used LITTLE-ENDIAN when writing. xD

**Comment by jython234** [comment]
@DefinitlyEvil Ok, i can't seem to find where in the code you are writing all this stuff.

**Comment by DefinitlyEvil** [comment]
@jython234 PCMultiChunkDataPacketTranslator

**Comment by jython234** [comment]
@DefinitlyEvil Yeah IDK, it could be something with your NBT lib.

**Comment by DefinitlyEvil** [comment]
@jython I used Nukkit's and it is a port from PocketMine-MP.

**Comment by DefinitlyEvil** [comment]
@DragonetMC/developers So people could u guys please focus on this issue? This is VERY important because if we get NBT, we can totally support ALL items on PC by adding our custom NBT tags to identify each item.

**Comment by jython234** [comment]
@DefinitlyEvil yeah Idk it should work

**Comment by DefinitlyEvil** [comment]
@jython234 Must be some packet encoding issue?

**Comment by jython234** [comment]
@DefinitelyEvil it could be, but the encoding looks right, maybe it is with
whatever you are putting into nbt?

**Comment by DefinitlyEvil** [comment]
@jython234 I wanted to put block entity data such as sign/chest/wallsign NBT data.

**Comment by DefinitlyEvil** [comment]
@MrPowerGamerBR I saw you solved the NBT issue in ProtocolSupport, couldu plz help to integrate it? xD

**Comment by MrPowerGamerBR** [comment]
@DefinitlyEvil what kind of issue did I solve on ProtocolSupport? IIRC I only made block/item remapper commits.

**Comment by MrPowerGamerBR** [comment]
@DefinitlyEvil Unless you are talking about the enchanted items NBT bug (but that is for the PC version) which Shevchik fixed it by looping the NBT and removing the enchanted id.

**Comment by DefinitlyEvil** [comment]
@MrPowerGamerBR No, NBT is totally not working in DragonProxy. For example, sign can't be displayed even with correct packet sent.

**Comment by MrPowerGamerBR** [comment]
@DefinitlyEvil yeah, but what did I do related to NBT on ProtocolSupport?

It is not like I'm the powerful god of NBT packets. :stuck_out_tongue: 

Also, isn't the sign NBT from MCPE different than the MCPC one?

**Comment by DefinitlyEvil** [comment]
@MrPowerGamerBR NBT in MCPE uses little-endian, that's the only difference in format.

**Comment by MrPowerGamerBR** [comment]
@DefinitlyEvil well, then see how the god of the MCPE packets (Shevchik) done this.

**Comment by DefinitlyEvil** [comment]
@MrPowerGamerBR Will do. ;P
